Add OOM killer info (#772)
timescale/docs: cloud/memory-management.md

# Memory management
If you run intensive queries on your Timescale Cloud services, you might
encounter out of memory (OOM) errors. This occurs if your query consumes more
memory than is available.

> When this happens, an `OOM killer` process shuts down PostgreSQL processes using
> `SIGKILL` commands, until the memory usage falls below the upper limit. To
> prevent service disruption caused by OOM errors, Timescale Cloud attempts to
> shut down only the query that caused the problem. This means that the
> problematic query does not run, but that your PostgreSQL service continues to
> operate normally.
> 
> OOM errors can kill the entire server process, and usually require a restart.
> When the OOM killer is triggered, you need to wait for the entire service to
> come back online before you can reconnect. The error in the log usually looks
> like this:

Reviewer: I got confused jumping back and forth between Timescale's way and the normal OOM way. One possible way to clarify:

```suggestion
When this happens, an `OOM killer` process shuts down PostgreSQL processes using
`SIGKILL` commands, until the memory usage falls below the upper limit. Because
this kills the entire server process, it usually requires a restart. To
prevent service disruption caused by OOM errors, Timescale Cloud attempts to
shut down only the query that caused the problem. This means that the
problematic query does not run, but that your PostgreSQL service continues to
operate normally.

If the normal OOM killer is triggered, the error log looks like this:
```

```

Wait for the entire service to come back online before reconnecting.

If Timescale Cloud successfully guards the service against the OOM killer, it shuts
down only the client connection that was using too much memory. This prevents 

> the entire PostgreSQL service from being shut down, so you can reconnect 

Author: ```suggestion
the entire PostgreSQL service from shutting down, so you can reconnect 
```

> PostgreSQL service from being shut down, so you can reconnect immediately. The

Author: ```suggestion
PostgreSQL service from shutting down, so you can reconnect immediately. The
```

> Timescale Cloud guards the service against the OOM killer by only shutting down
> the client connection which was using too much memory. This prevents the entire
> PostgreSQL service from being shut down, so you can reconnect immediately. The
> error in the log for Timescale Cloud looks like this:

Reviewer: Suggestion continued:

```suggestion
Wait for the entire service to come back online before reconnecting.

If Timescale Cloud successfully guards the service against the OOM killer, it shuts
down only the client connection that was using too much memory. This prevents 
the entire PostgreSQL service from being shut down, so you can reconnect 
immediately. The error log looks like this:
```
